### PR TITLE
FlightSqlProducer - Refactor DremioFlightProducer to delegate between FlightSql and legacy producers

### PR DESCRIPTION
**Pull request by rafael-telles**
This refactors DremioFlightProducer to use a new FlightSqlProducer implementation (to be implemented in other PRs)

### CODE REVIEW

File: services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java [hunk 1]
```diff
@@ -113,4 +157,36 @@ public void listActions(CallContext callContext, StreamListener<ActionType> stre
   private CallHeaders retrieveHeadersFromCallContext(CallContext callContext) {
     return callContext.getMiddleware(FlightConstants.HEADER_KEY).headers();
   }
+
+  // TODO: Add this to FlightSqlUtils
```
Comment (kylep-dremio) [inline services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java:161]: This should be done before I'd consider approving...

Comment (rafael-telles) [reply services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java:161]: Fixed - https://github.com/rafael-telles/arrow-experimental/pull/74/files

File: services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java [hunk 2]
```diff
@@ -113,4 +348,34 @@ public void listActions(CallContext callContext, StreamListener<ActionType> stre
...
+    } catch (InvalidProtocolBufferException e) {
+      return false;
+    }
+  }
+
+  private boolean isFlightSqlCommand(FlightDescriptor flightDescriptor) {
+    return isFlightSqlCommand(flightDescriptor.getCommand());
+  }
+
+  private boolean isFlightSqlCommand(Ticket ticket) {
```
Comment (jduo) [inline services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java:373]: I don't think I follow this. When can a ticket ever be a FlightSqlCommand?

File: services/arrow-flight/src/test/java/com/dremio/service/flight/FlightClientUtils.java [hunk 3]
```diff
@@ -36,17 +37,23 @@
   public static final class FlightClientWrapper implements AutoCloseable {
     private BufferAllocator allocator;
```
Comment (jduo) [inline services/arrow-flight/src/test/java/com/dremio/service/flight/FlightClientUtils.java:38]: Make these fields final

Comment (rafael-telles) [reply services/arrow-flight/src/test/java/com/dremio/service/flight/FlightClientUtils.java]: Fixed

File: services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java (hunk 2 above)
Comment (rafael-telles) [reply services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java]: Fixed - actually the ticket bytes are a serialized FlightSqlCommand, renamed and added comments

**Review by jduo — APPROVED** [review]

**Comment by rafael-telles** [comment]
Just closed because this is our base branch

**Pull request closed without merging**